urllib3/util/response.py

```diff
@@ -20,3 +37,34 @@ def is_fp_closed(obj):
         pass
 
     raise ValueError("Unable to determine whether fp is closed.")
+
+
+def extract_parsing_errors(headers):
+    """
+    Extracts encountered errors from the result of parsing headers.
+
+    Only works on Python 3 (and PyPy 2).
+
+    :param headers: Headers to extract errors from.
+    :type headers: `httplib.HTTPMessage`.
+
+    :returns: Defects encountered while parsing headers and unparsed header
+              data.
+    :rtype: HeaderParsingErrors
+    """
+
+    # This will fail silently if we pass in the wrong kind of parameter.
+    # To make debugging easier add an explicit check.
+    if not isinstance(headers, httplib.HTTPMessage):
+        raise TypeError('expected httplib.Message, got {}.'.format(
+            type(headers)))
+
+    defects = getattr(headers, 'defects', None)
+    get_payload = getattr(headers, 'get_payload', None)
+
+    if get_payload:  # Platform-specific: Implementation dependent.
+        unparsed_data = get_payload()
+    else:  # Platform-specific
```

Comment:
Can we set `unparsed_data = None` before the if-statement and just let the default pass-through? (Not a fan of else's that don't do anything)